src/sentry/static/sentry/app/utils/errorHandler.jsx

```diff
@@ -2,17 +2,32 @@ import React from 'react';
 import RouteError from 'app/views/routeError';
 
 export default function errorHandler(Component) {
-  const originalRender = Component.prototype.render;
-  Component.prototype.render = function() {
-    try {
-      return originalRender.apply(this, arguments);
-    } catch (err) {
-      /*eslint no-console:0*/
+  class ErrorHandler extends React.Component {
+    static getDerivedStateFromError(error) {
       setTimeout(() => {
-        throw err;
+        throw error;
```

Comment:
Hmm wonder if we even need this since `RouteError` captures errors to Sentry

Reply:
@billyvg I tried preserving the implementation as much as I could; since I'm not aware if anything is catching this re-thrown error.

Looks like this PR introduced this `setTimeout()`: https://github.com/getsentry/sentry/pull/4443

Reply:
Makes sense. I don't think there's anything catching it - I think we only use `errorHandler` in our routes, so it would probably only bubble up to `app/main` and `app/index`. Can we follow up with a PR to see if we remove it? I would say if we throw an error in a component and we have a reasonable stacktrace, it can go.

Reply:
Yea, I'll follow up with a PR to remove it.